modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/GroupPartitionId.java

```diff
@@ -0,0 +1,160 @@
+    /** Index for super(meta) page. There is always such page for iterated cache partition. */
+    private static final int METAPAGE_IDX = 0;
+
+    /** Group ID. */
+    private final int grpId;
+
+    /** Partition ID. */
+    private final int partId;
+
+    /**
+     * Creates group-partition tuple.
+     *
+     * @param grpId Group ID.
+     * @param partId Partition ID.
+     */
+    public GroupPartitionId(final int grpId, final int partId) {
+        this.grpId = grpId;
+        this.partId = partId;
+    }
+
+    /**
+     * Returns flag to be used for partition.
+     *
+     * @param partId Partition ID.
+     */
+    public static byte getFlagByPartId(final int partId) {
+        return partId == INDEX_PARTITION ? FLAG_AUX : FLAG_DATA;
+    }
+
+    /**
+     * Return group ID.
+     */
+    public int getGroupId() {
+        return grpId;
+    }
+
+    /**
+     * Return partition ID.
+     */
+    public int getPartitionId() {
+        return partId;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public String toString() {
+        return S.toString(GroupPartitionId.class, this);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        GroupPartitionId key = (GroupPartitionId) o;
+
+        if (grpId != key.grpId) {
+            return false;
+        }
+
+        return partId == key.partId;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public int hashCode() {
+        int result = grpId;
+
+        result = 31 * result + partId;
+
+        return result;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public int compareTo(GroupPartitionId o) {
+        if (getGroupId() < o.getGroupId()) {
+            return -1;
+        }
+
+        if (getGroupId() > o.getGroupId()) {
+            return 1;
+        }
+
+        if (getPartitionId() < o.getPartitionId()) {
+            return -1;
+        }
+
+        if (getPartitionId() > o.getPartitionId()) {
+            return 1;
+        }
+
+        return 0;
+    }
+
+    /**
+     * Returns page ID (64 bits) constructed from partition ID and given index.
+     *
+     * @param pageIdx Page Index, monotonically growing number within each partition.
+     */
+    private long createPageId(final int pageIdx) {
+        final int partId = getPartitionId();
+
+        return pageId(partId, getFlagByPartId(partId), pageIdx);
+    }
+
+    /**
+     * Returns Full page ID. For index 0 will return super-page of next partition
+     *
+     * @param pageIdx Page Index, monotonically growing number within each partition
+     * @return FullPageId consists of cache ID (32 bits) and page ID (64 bits).
+     */
+    private FullPageId createFullPageId(final int pageIdx) {
+        return new FullPageId(createPageId(pageIdx), getGroupId());
```

Comment:
If it's all just for meta page id, then we could just hardcode these expressions into "createFirstPageFullId", if it's even used

Reply:
The code is not needed got rid of it.